### DISCUSSION TITLE
Amazon.IdentityStore CreateUser Not Sending Email Verification Link

### DISCUSSION THREAD

**Discussion by feardobeardo2** [aws/aws-sdk-net#3288] [Q&A] [answered]
### Describe the bug

When creating a user with the Amazon.IdentityStore API the user does not receive the email verification, we have to go in and manually send it. Not really seeing the point of this API if the end user cannot verify to login. I have also tried enabling **Send email OTP** just to see if it works and it does not. Only way for the user to get the email is by clicking the button in the IAM Identity Center console.

### Expected Behavior

Create user via API, email is sent to user so they can login. 

### Current Behavior

Create User via API, no email is sent to the user.

### Reproduction Steps

**CODE SAMPLE**

```
public Amazon.IdentityStore.Model.CreateUserResponse CreateIdentityStoreUser( string emailAddress, string displayName, string firstName, string lastName )
    {
      var identityService = new AmazonIdentityStoreClient( AccessKeyID, accessKeySecret, Amazon.RegionEndpoint.GetBySystemName( RegionEndpoint ) );
      var createUserRequest = new Amazon.IdentityStore.Model.CreateUserRequest
      {
        IdentityStoreId = identityStoreId,
        UserName = emailAddress,
        Emails = new List<Amazon.IdentityStore.Model.Email>
        {
          new Amazon.IdentityStore.Model.Email()
          {
             Primary = true,
             Type = "Work",
             Value = emailAddress
          }
        },
        DisplayName = displayName,
        Name = new Amazon.IdentityStore.Model.Name()
        {
          FamilyName = lastName,
          GivenName = firstName
        }
      };
      var createUserResponse = identityService.CreateUser( createUserRequest );

      return createUserResponse;

    }
```

### AWS .NET SDK and/or Package version used

AWSSDK.IdentityStore 3.7.300.68

### Targeted .NET Platform

net452

### Operating System and version

Windows

**Comment by ashishdhingra** [comment]
`AWSSDK.IdentityStore` package is autogenerated from service model. Most likely a service API or configuration issue. Needs reproduction.

**Comment by ashishdhingra** [comment, marked as answer]
@feardobeardo2 Good afternoon. Thanks for reporting the issue. This appears to be reproducible using code below:
```C#
using Amazon;
using Amazon.IdentityStore;
using Amazon.IdentityStore.Model;

var response = await CreateIdentityStoreUser("d-<<some-id>>", RegionEndpoint.USEast1, "testemail@testdomain.com", "Test User", "Test", "User");
Console.WriteLine(response.UserId);
async Task<CreateUserResponse> CreateIdentityStoreUser(string identityStoreId, RegionEndpoint regionEndpoint, string emailAddress, string displayName, string firstName, string lastName)
{
    var identityService = new AmazonIdentityStoreClient(regionEndpoint);
    var createUserRequest = new CreateUserRequest
    {
        IdentityStoreId = identityStoreId,
        UserName = emailAddress,
        Emails = new List<Email>
        {
          new Email()
          {
             Primary = true,
             Type = "Work",
             Value = emailAddress
          }
        },
        DisplayName = displayName,
        Name = new Name()
        {
            FamilyName = lastName,
            GivenName = firstName
        }
    };
    return await identityService.CreateUserAsync(createUserRequest);
}
```
After user is successfully created, examining user details in AWS console displays below banner at the top:
```
Email not verified
Users must first verify their email address before they can begin to use certain features such as completing email-based two-step verification during sign-in.
```

There doesn't appear to be any service API operation per [Identity Store API Reference](https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/welcome.html) to send email verification link to user.

Upon investigating further (thanks for article [Add a layer of security for AWS IAM Identity Center user portal sign-in with context-aware email-based verification](https://aws.amazon.com/blogs/security/add-a-layer-of-security-for-aws-sso-user-portal-sign-in-with-context-aware-email-based-verification/)), here are the additional steps required:
- Enable email OTP for standard authentication.
![Identity Store - Email Verification setting](https://github.com/aws/aws-sdk-net/assets/67916761/0f2e02d6-a048-4f5f-b47c-5db5d392f7d0)
- After user is created using `CreateUser` API (i.e. using above code), they can try login using `AWS access portal URL`. This is available from Identity Center settings and looks something like `https://<<identity-store-id>>.awsapps.com/start`.
- Once user tries to login using their registered email address, they would be prompted to verify email using verification code. This is when the verification email with OTP would be sent.
  - Once verified, user would be prompted to setup new password and username.

So `CreateUser` only creates users and depending on the Identity Store setting, verification link is sent to users while they try to sign-in for the first time.

I was able to test the above flow at my end and verified that it works successfully.

Hope this helps.

Thanks,
Ashish

**Comment by chgogu** [comment]
Hello @ashishdhingra  

We are creating the AWS Identity Store user via createUserRequest Java API and seeing recently that user email is not verified. Can you please let us know if there is any work around for this issue? are there any new API's available to send the invitation or verification email or verify the email while creating user via API? 

Thanks,
Chaitanya
